app/helpers/application_helper.rb

```diff
@@ -16,4 +16,17 @@ def link_to_add_polymorphic_association(*args)
     html_options[:partial] = "#{association}/fields"
     link_to_add_association(name, f, association, html_options)
   end
+
+  def form_navigation_btn(action)
+    text = action == :back ? t('back') : t_title(action)
+    action = :index if action == :back
+    target = { controller: controller_name, action: action }
+    button_type = action == :new ? 'success' : 'default'
+    target[:id] = params[:id] unless action == :index
```

Comment:
I couldn't totally understand the single line conditionals from line 21 - 25,
can we have a look together on this